zcash_client_backend/src/keys.rs

```diff
@@ -28,18 +45,223 @@ pub fn spending_key(seed: &[u8], coin_type: u32, account: u32) -> ExtendedSpendi
         &[
             ChildIndex::Hardened(32),
             ChildIndex::Hardened(coin_type),
-            ChildIndex::Hardened(account),
+            ChildIndex::Hardened(account.0),
         ],
     )
 }
 
+#[cfg(feature = "transparent-inputs")]
+pub fn derive_transparent_address_from_secret_key(
+    secret_key: &secp256k1::key::SecretKey,
+) -> TransparentAddress {
+    let secp = Secp256k1::new();
+    let pk = PublicKey::from_secret_key(&secp, secret_key);
+    derive_transparent_address_from_public_key(&pk)
+}
+
+#[cfg(feature = "transparent-inputs")]
+pub fn derive_transparent_address_from_public_key(
+    public_key: &secp256k1::key::PublicKey,
+) -> TransparentAddress {
+    let mut hash160 = ripemd160::Ripemd160::new();
+    hash160.update(Sha256::digest(&public_key.serialize()));
+    TransparentAddress::PublicKey(*hash160.finalize().as_ref())
+}
+
+#[cfg(feature = "transparent-inputs")]
+pub fn derive_secret_key_from_seed<P: consensus::Parameters>(
+    params: &P,
+    seed: &[u8],
+    account: AccountId,
+    index: u32,
+) -> Result<SecretKey, hdwallet::error::Error> {
+    let private_key =
+        derive_extended_private_key_from_seed(params, seed, account, index)?.private_key;
+    Ok(private_key)
+}
+
+#[cfg(feature = "transparent-inputs")]
+pub fn derive_public_key_from_seed<P: consensus::Parameters>(
+    params: &P,
+    seed: &[u8],
+    account: AccountId,
+    index: u32,
+) -> Result<PublicKey, hdwallet::error::Error> {
+    let private_key = derive_extended_private_key_from_seed(params, seed, account, index)?;
+    let pub_key = ExtendedPubKey::from_private_key(&private_key);
+    Ok(pub_key.public_key)
+}
+
+#[cfg(feature = "transparent-inputs")]
+pub fn derive_extended_private_key_from_seed<P: consensus::Parameters>(
+    params: &P,
+    seed: &[u8],
+    account: AccountId,
+    index: u32,
+) -> Result<ExtendedPrivKey, hdwallet::error::Error> {
+    let pk = ExtendedPrivKey::with_seed(&seed)?;
+    let private_key = pk
+        .derive_private_key(KeyIndex::hardened_from_normalize_index(44)?)?
+        .derive_private_key(KeyIndex::hardened_from_normalize_index(params.coin_type())?)?
+        .derive_private_key(KeyIndex::hardened_from_normalize_index(account.0)?)?
+        .derive_private_key(KeyIndex::Normal(0))?
+        .derive_private_key(KeyIndex::Normal(index))?;
+    Ok(private_key)
+}
+
+#[cfg(feature = "transparent-inputs")]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Wif(pub String);
+
+#[cfg(feature = "transparent-inputs")]
+impl Wif {
+    pub fn from_secret_key(sk: &SecretKey, compressed: bool) -> Self {
+        let secret_key = sk.as_ref();
+        let mut wif = [0u8; 34];
+        wif[0] = 0x80;
+        wif[1..33].copy_from_slice(secret_key);
+        if compressed {
+            wif[33] = 0x01;
+            Wif(bs58::encode(&wif[..]).with_check().into_string())
+        } else {
+            Wif(bs58::encode(&wif[..]).with_check().into_string())
+        }
```

Comment:
I checked that this and the implementation of `bs58::encode(...).with_check()` at https://docs.rs/bs58/latest/src/bs58/encode.rs.html matches https://en.bitcoin.it/wiki/Wallet_import_format .

Reply:
Fixed, this now uses `params.wif_lead_byte()`